tests/testthat/test-sampler-random-walk.R

```diff
@@ -30,3 +30,18 @@ test_that("validate sampler against model on initialisation", {
     "Incompatible length parameters (1) and vcv (2)",
     fixed = TRUE)
 })
+
+
+test_that("can draw samples from a random model", {
+  set.seed(1)
+  m <- ex_dust_sir()
+  vcv <- matrix(c(0.0006405, 0.0005628, 0.0005628, 0.0006641), 2, 2)
+  sampler <- mcstate_sampler_random_walk(vcv = vcv)
+  res <- mcstate_sample(m, sampler, 20)
+  expect_setequal(names(res), c("pars", "density", "details", "chain"))
+  ## OK, this is ready then to start thinking about how we inject
+  ## observer support in here; we'll need access to the actual model
+  ## in order to get trajectories out etc, but that's not super
+  ## obvious really; that means we're going to need some sort of
+  ## special sauce for the observer function really.
+})
```

Comment:
I have not understood this comment well yet!

Reply:
ah, it's out of date now after #10

Reply:
(and now it's gone)

Reply:
would like to definitely chat through this and that sir model on call!